pkg/hubble/relay/observer.go

```diff
@@ -43,9 +45,14 @@ func (s *Server) GetFlows(req *observerpb.GetFlowsRequest, stream observerpb.Obs
 		}
 	}()
 
-	//TODO: figure out what makes a reasonnable channel size
 	peers := s.peerList()
-	flows := make(chan *observerpb.GetFlowsResponse, 10*len(peers))
+	//TODO: figure out what makes a reasonnable default size
+	qlen := 10 * len(peers)
+	if req.GetNumber() > 0 && req.GetNumber() < uint64(qlen) {
+		qlen = int(req.GetNumber()) * len(peers) // we collect Number x nPeer
```

Comment:
This seems a high number than required. We don't need to order everything it should be seen as a best effort.

Reply:
This implementation does not order everything (it would be impossible with follow mode anyway). The `qlen` variable is actually bounded to `10 x N` where `N` is the number of nodes.

Reply:
I've changed this logic a bit. With the change, the queue is now bounded to 100.